sdk/core/azure-core-amqp/inc/azure/core/amqp/message_receiver.hpp

```diff
@@ -112,13 +101,9 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     void Close();
     std::string GetLinkName() const;
     std::string GetSourceName() const;
-    uint32_t GetReceivedMessageId();
-    void SendMessageDisposition(
-        const char* linkName,
-        uint32_t messageNumber,
-        Models::AmqpValue deliveryState);
 
-    Models::AmqpMessage WaitForIncomingMessage(Context const& context = {});
+    std::pair<Azure::Nullable<Models::AmqpMessage>, Models::_internal::AmqpError>
```

Comment:
Should this be a type rather than a pair to provide for future compat?

Reply:
Maybe, but the same applies to all methods which return a single value.

Also, these types are all _internal, which doesn't have the same forward compatibility restrictions as public types.